app/message-table.tsx

```diff
@@ -132,6 +133,23 @@ export function MessageTable<TData, TValue>({
                               impressionsThreshold={impressionsThreshold}
                             />
                           )}
+                        {header.column.columnDef
+                          .header!.toString()
+                          .includes("Surface") && (
+                          <div>
+                            <input
+                              type="text"
+                              value={
+                                (header.column.getFilterValue() ?? "") as string
+                              }
+                              onChange={(e) =>
+                                header.column.setFilterValue(e.target.value)
+                              }
+                              placeholder={`Search...`}
+                              className="w-full border border-slate-400 font-light p-1 text-2xs rounded"
+                            />
+                          </div>
+                        )}
```

Comment:
For maintainability & readability, how about hoisting this into its own functional component in this file.